src/primitives/transaction.h

```diff
@@ -331,12 +331,22 @@ class CCoinsView;
 /** Precompute sighash midstate to avoid quadratic hashing */
 struct PrecomputedTransactionData {
     uint256 hashPrevouts, hashSequence, hashOutputs;
+    /** Coins spend by the tx */
     std::vector<CTxOut> m_spent_outputs;
-    uint256 m_prevouts_single_hash;
-    uint256 m_sequences_single_hash;
-    uint256 m_outputs_single_hash;
-    uint256 m_spent_amounts_single_hash;
-    uint256 m_spent_scripts_single_hash;
+    /** Merkle root of the H(input.prevout || input.nSequence) for all inputs */
+    uint256 m_inputs_merkle_root;
+    /** Merkle root of the coins spend by the tx */
+    uint256 m_inputs_spent_outputs_merkle_root;
```

Comment:
👍 I was thinking we want to make sure not to rehash all these if someone specifies these flags.